frontend/packages/ceph-storage-plugin/src/components/common/noobaa-store/bs-ns-common.tsx

```diff
@@ -0,0 +1,267 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { Dropdown, Firehose } from '@console/internal/components/utils';
+import { Button, FormGroup, TextInput, InputGroup } from '@patternfly/react-core';
+import { SecretModel } from '@console/internal/models';
+import { ResourceDropdown, getAPIVersion } from '@console/shared';
+import './create-bs-ns.scss';
+import { SecretType } from '@console/internal/components/secrets/create-secret';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+import { BC_PROVIDERS, AWS_REGIONS } from '../../../constants';
+
+export enum StoreType {
+  BS = 'BackingStore',
+  NS = 'NamespaceStore',
+}
+export const awsRegionItems = _.zipObject(AWS_REGIONS, AWS_REGIONS);
+export const endpointsSupported = [BC_PROVIDERS.S3, BC_PROVIDERS.IBM];
+export const getProviders = (type: StoreType) => {
+  const values =
+    type === StoreType.BS
+      ? Object.values(BC_PROVIDERS)
+      : Object.values(BC_PROVIDERS).filter(
+          (provider) => provider !== BC_PROVIDERS.GCP && provider !== BC_PROVIDERS.PVC,
+        );
+  return _.zipObject(values, values);
+};
+
+export const getExternalProviders = (type: StoreType) => {
+  return type === StoreType.NS
+    ? [BC_PROVIDERS.AWS, BC_PROVIDERS.AZURE, BC_PROVIDERS.S3, BC_PROVIDERS.IBM]
+    : [BC_PROVIDERS.AWS, BC_PROVIDERS.AZURE, BC_PROVIDERS.S3, BC_PROVIDERS.GCP, BC_PROVIDERS.IBM];
+};
```

Comment:
These all can go to `utils/noobaa-utils.ts`

Reply:
done